discord_bot.py

```diff
@@ -1,282 +1,297 @@
-    client = MyClient(state_path, config)
-    client.run(os.environ['DISCORD_TOKEN'])
-
-
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser()
-    parser.add_argument('--config', type=Path, default='config.json')
-    parser.add_argument('--state', type=Path, default='state')
-    parser.add_argument('--log-console', action='store_true')
-    args = parser.parse_args()
-
-    if args.log_console:
-        logging.basicConfig(
-            level=logging.DEBUG,
-            format='%(asctime)s %(levelname)s %(filename)s:%(lineno)s - %(message)s'
-        )
-
-    else:
-        logging.basicConfig(
-            level=logging.DEBUG,
-            filename=LOG_FILE,
-            format='%(asctime)s %(levelname)s %(filename)s:%(lineno)s - %(message)s'
-        )
-
-    config = json.loads(args.config.read_text())
-
-    main(args.state, config)
+import asyncio
+import os
+from typing import TypedDict
+
+from metrics import MetricsHandler
+
+
+def load_env():
+    with open('secrets.env') as file:
+        for line in file:
+            line = line.strip()
+            if not line or line.startswith('#'):
+                continue
+
+            key, value = line.split('=')
+            os.environ[key] = value
+
+
+load_env()
+
+import argparse
+import json
+import logging
+logging.basicConfig(level=logging.DEBUG)
+from pathlib import Path
+
+import discord
+
+from rubber_duck import Message, RubberDuck, MessageHandler, ErrorHandler, Attachment
+from quest import create_filesystem_manager
+from bot_commands import BotCommands
+
+LOG_FILE = Path('/tmp/duck.log')  # TODO - put a timestamp on this
+
+
+def parse_blocks(text: str, limit=2000):
+    tick = '`'
+    block = ""
+    current_fence = ""
+    for line in text.splitlines():
+        if len(block) + len(line) > limit - 3:
+            if block:
+                if current_fence:
+                    block += '```'
+                yield block
+                block = current_fence
+
+        block += ('\n' + line) if block else line
+
+        if line.strip().startswith(tick * 3):
+            if current_fence:
+                current_fence = ""
+            else:
+                current_fence = line
+
+    if block:
+        yield block
+
+
+def as_message(message: discord.Message) -> Message:
+   
+    return Message(
+        guild_id=message.guild.id,
+        channel_name=message.channel.name,
+        channel_id=message.channel.id,
+        author_id=message.author.id,
+        author_name=message.author.name,
+        author_mention=message.author.mention,
+        message_id=message.id,
+        
+        content=message.content,
+        # is_file=len(message.attachments) > 0
+
+        file = [as_attachment(attachment) for attachment in message.attachments] # call new as_attachment
+    )
+#create function that takes a attachment and return a dictionary
+#converts each attachment to a dictionary try __dict()
+# get filename and id
+

```

Comment:
Make sure your IDE has a pep8 formatter installed (VSCode and PyCharm both do). Use it before you commit your code.